### PR TITLE
QL: add redundant-assignment query

### PR DESCRIPTION
**Pull request by erik-krogh**
A quick idea I had.  
Assigning the same value to a variable twice is redundant, so the second assignment can be removed.  
There weren't a lot of results.

### CODE REVIEW

**Review by owen-mc — APPROVED** [review]
Go changes approved

**Review by hvitved — APPROVED** [review]
C# 👍  Nice!

File: ql/ql/src/queries/style/RedundantAssignment.ql [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * Holds if `assigned1` and `assigned2` assigns the same value to `var`.
+ * The assignments may be on different branches of a disjunction.
+ */
+predicate candidateRedundantAssignment(AssignedVariable var, Expr assigned1, Expr assigned2) {
+  assigned1 = var.getAnAssignedExpr() and
+  assigned2 = var.getAnAssignedExpr() and
+  (
+    GVN::valueNumber(assigned1) = GVN::valueNumber(assigned2)
+    or
+    // because GVN skips large strings, we need to check for equality manually
```
Comment (MathiasVP) [inline ql/ql/src/queries/style/RedundantAssignment.ql:55]: I really should investigate why this ugly hack is necessary 🤔.

Comment (erik-krogh) [reply ql/ql/src/queries/style/RedundantAssignment.ql:55]: I think it's the MaD rows that go wrong in some way.
There are some massive disjunctions in there.

**Pull request merged**